Added overview of C# member access operators

pkulikov opened the pull request:
Fixes #11118
Supported by dotnet/samples#886

pkulikov committed:
Added overview of C# member access operators

pkulikov committed:
Updated TOC

pkulikov committed:
Use indexer operator

pkulikov committed:
Minor edits

pkulikov committed:
Setup redirection

pkulikov committed:
Minor edits

BillWagner reviewed (changes requested):
Super great work @pkulikov 

I had one item for discussion.  I'm marking as "request changes" so it stays in my review queue. Checking the build status and review site next.

BillWagner commented:
Here's the [internal review link](https://review.docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators?branch=pr-en-us-12240)

pkulikov committed:
Addressed feedback

pkulikov commented on docs/csharp/language-reference/operators/member-access-operators.md:
```diff
@@ -0,0 +1,163 @@
+
+Available in C# 6 and later, a null-conditional operator applies a member access, `?.`, or element access, `?[]`, operation to its operand only if that operand evaluates to non-null. If the operand evaluates to `null`, the result of applying the operator is `null`.
+
+The null-conditional operators are short-circuiting. That is, if one operation in a chain of conditional member or element access operations returns `null`, the rest of the chain doesn't execute. In the following example, `E` doesn't execute if `A`, `B`, or `C` evaluates to `null`:
```
@BillWagner agree with you; I've made the chain shorter and updated the wording. Please take a look at the updated text.

BillWagner replied:
I think it is worth adding that if `B` is `null`, `C` is not evaluated. (as an example of the short circuiting).

pkulikov commented:
@BillWagner thanks for review! I've addressed your feedback.

(Strangely, your review comments are timestamped into the future (at least on my side))

BillWagner commented:
Thanks again @pkulikov 

I'll :shipit: now.

BillWagner commented on docs/csharp/language-reference/operators/member-access-operators.md:
```diff
@@ -0,0 +1,163 @@
+
+The preceding example also shows the usage of the [null-coalescing operator](null-coalescing-operator.md). You might use the null-coalescing operator to provide an alternative expression to evaluate in case the result of the null-conditional operation is `null`.
+
+### Thread-safe delegate invocation
```
I'm so glad you added this example.
